sdk/core/azure-core/src/main/java/com/azure/core/http/policy/BearerTokenAuthenticationPolicy.java

```diff
@@ -33,22 +36,78 @@
      */
     public BearerTokenAuthenticationPolicy(TokenCredential credential, String... scopes) {
         Objects.requireNonNull(credential);
-        Objects.requireNonNull(scopes);
-        assert scopes.length > 0;
         this.credential = credential;
         this.scopes = scopes;
-        this.cache = new SimpleTokenCache(() -> credential.getToken(new TokenRequestContext().addScopes(scopes)));
+        this.cache = new AccessTokenCache(credential);
+    }
+
+    /**
+     * Executed before sending the initial request and authenticates the request.
+     *
+     * @param context The request context.
+     * @return A {@link Mono} containing {@link Void}
+     */
+    public Mono<Void> authorizeRequest(HttpPipelineCallContext context) {
```

Comment:
Since this is a policy plugged into the pipeline - who will call this method publicly?

Reply:
So, the expectation is that, any SDK looking to add their custom authorization logic for the first/initial request will override this method. 
An example would be KeyVault SDK, as their initial request is sent unauthenticated, their auth policy will need to override this method and not do any authorization.
Another example would ARM SDK, where they would override this method, determine their scopes on the fly, create their custom TRC and authorize the request using it.